File: scripts_py/leap_year.py
def is_leap_year(year):
    if (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0):
        return True
    else:
        return False

# Получаем год от пользователя
year = int(input("Введите год: "))

# Проверяем, является ли год високосным, и выводим результат
if is_leap_year(year):
    print(f"{year} год - високосный.")
else:
    print(f"{year} год - не високосный.")

File: scripts_c++/leap_year.cpp
#include <iostream>
bool isLeapYear(int year) {
    return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0);
}
int main() {
    int year;
    std::cout << "Введите год для проверки на високосность: ";
    std::cin >> year;
    if (isLeapYear(year)) {
        std::cout << year << " - високосный год." << std::endl;
    } else {
        std::cout << year << " - не високосный год." << std::endl;
    }

    return 0;
}